.github/scripts/upload_tests_results.py

```diff
@@ -0,0 +1,342 @@
+    tree = ET.parse(xml_file)
+    root = tree.getroot()
+
+    results = []
+    for testsuite in root.findall(".//testsuite"):
+        suite_name = testsuite.get("name")
+
+        for testcase in testsuite.findall("testcase"):
+            name = testcase.get("name")
+            rawname, filename, testname, fixture = parse_test_name(name)
+            time = testcase.get("time")
+
+            status_description = ""
+            status = "passed"
+            if testcase.find("properties/property/[@name='mute']") is not None:
+                status = "mute"
+                status_description = testcase.find(
+                    "properties/property/[@name='mute']"
+                ).get("value")
+            elif testcase.find("failure") is not None:
+                status = "failure"
+                status_description = testcase.find("failure").text
+            elif testcase.find("error") is not None:
+                status = "error"
+                status_description = testcase.find("error").text
+            elif testcase.find("skipped") is not None:
+                status = "skipped"
+                status_description = testcase.find("skipped").text
+
+            results.append(
+                {
+                    "build_type": build_type,
+                    "commit": commit,
+                    "branch": branch,
+                    "pull": pull,
+                    "run_time": run_time,
+                    "job_name": job_name,
+                    "job_id": job_id,
+                    "suite_name": suite_name,
+                    "raw_test_name": rawname,
+                    "file": filename,
+                    "test_name": testname,
+                    "fixture": fixture,
+                    "time": float(time),
+                    "status": status,
+                    "status_description": status_description.replace("\r\n", ";;").replace("\n", ";;").replace("\"", "'"),
+                    "log": "" if testcase.find("properties/property/[@name='url:log']") is None else testcase.find("properties/property/[@name='url:log']").get('value'),
+                    "logsdir": "" if testcase.find("properties/property/[@name='url:logsdir']") is None else testcase.find("properties/property/[@name='url:logsdir']").get('value'),
+                    "stderr": "" if testcase.find("properties/property/[@name='url:stderr']") is None else testcase.find("properties/property/[@name='url:stderr']").get('value'),
+                    "stdout": "" if testcase.find("properties/property/[@name='url:stdout']") is None else testcase.find("properties/property/[@name='url:stdout']").get('value'),
+
+                }
+            )
+    return results
+
+
+def get_codeowners_for_tests(entries, tests_data):
+    tests_data_with_owners = []
+    for test in tests_data:
+        owners = []
+        target_path = test["suite_name"]
+        for path, path_owners in entries:
+            path = path + "/" if path[-1] != "/" else path
```

Comment:
Will this code work with cases like 
```
*.js    @user
```
? 

Also probably it is a good idea to use a specialized lib for that job (5 seconds of googling: https://github.com/sbdchd/codeowners )